sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -814,12 +868,17 @@ def _plot_ice_lines(
             n_ice_to_plot,
             replace=False,
         )
+
+        individual_line_kw = ice_lines_kw.copy()
+        ice_lines_kw = self._check_if_colors_by_feature(ice_lines_kw, preds)
+        color_list = ice_lines_kw["color_dict"]["color_list"]
```

Comment:
iwie ist das ein ziemliches hin und her an welcher stelle du was brauchst... also hier brauchst du ja eigentlich nur die `color_list`, an anderer stelle dann die legend info. und dann speicherst du die sachen auch immer iwo in dem dict ab aber popst sei dann auch wieder raus .... vllt kann deine funktion einfach beides zurück geben (liste und legend kram) und an der stelle hier ignorierst du den legend kram und nimmst nur die liste (ohne sie iwo in ein dict zu packen) und in der anderen funktion unten rufst du das nochmal auf und nimmst da dann die legend infos?